### ISSUE TITLE
API incorporation request

### ISSUE THREAD

**Issue by starlilyth** [cbuchner1/CudaMiner#139]
I would like to request the incorporation of an API for external monitoring and control of CudaMiner, in the style of CGMiner (https://github.com/ckolivas/cgminer/blob/3.7/API-README). 

I understand this is not a trivial feature, however, the benefits of having CudaMiner manageable with pre-existing tools are many. As a developer of mining interface applications, I can tell you that "make PoolManager work with CudaMiner" is a very frequent request from users. 

This can be implemented in stages, with basic version and status reporting as a start. Please let me know if I can be of any assistance. 

Thanks, Lily
https://miner.farm

**Comment by cbuchner1** [comment]
Yes this was planned at one point, but the looming demise of scrypt GPU
mining prompted us to put priority into the ccMiner application. Currently
X11 mining is fastest on nVidia GPUs because of our work.

Because both programs share a lot of code, I think any attempt to
incorporate an API into one codebase would also make merging into the other
codebase easy.

The API work was not forgotten, just delayed a bit.

Christian

2014-05-19 17:59 GMT+02:00 Lily Star notifications@github.com:

**Comment by melt7777** [comment]
Thank you guys, we are seeing a ton of requests which would require an API for cudaminer and ccMiner, and I am fine with priority being placed there and then slide it into cudaminer too. Anything to move this along would be supremely helpful to anyone with an nvidia mining operation, which is growing exponentially.
